### ISSUE TITLE
[Discussion] Using Connected nodes for wiring.

### ISSUE THREAD

**Issue by C1ffisme** [Jeija/minetest-mod-mesecons#266]
I wanted to open this for discussion and thought.

So, why would we change the wires to connected nodes?

- All the normal advantages with less nodes

We don't need to worry about horrendous loading times at startup or anything else. Using a connected node, we only use 2 nodes: one for on and one for off.

- More possibilities for wire types

We could make it possible to have different types of wires, such as different colors or different levels of power. (Similar to redstone's "analog output", see [here](http://minecraft.gamepedia.com/Redstone#Redstone_component) for more information on that.)

The only problem though, is that it would require a lot of time and effort. Here's what would need to be done:

- Legacy Support for older worlds. (LOTS of aliases.)

- A new system of wire detection that works with nodes of only one type. (Already done with mese nodes, though.)

Tell me if I missed anything.

**Comment by sofar** [comment]
mesecons wires connect to potentially 9(10?) neighboring nodes instead of the 6 that connected nodes supports. It would be an improvement, but it would break a lot of existing circuitry.

If you'd try and eliminate nodes, it can still be useful, though, but it won't ever catch all of them in one node.

**Comment by VanessaE** [comment]
Seems to me that that calls for an upgrade to the way the engine does connected nodes.  :smile: 

FWIW, a regular wire can connect to any or all of 12 possible neighbors:  the four ±X,±Z faces, the four ±X/±Y paths, and the four ±Z/±Y paths.  To go vertical, a separate kind of mesecons wire is provided (conducts sideways or turns the signal to/from a vertical path, does not connect to horizontal neighbors except via the plate that appears at the very top/bottom of the run).

A mese block passes signals to/from wires only at its sides (including the top/bottom plate of a vertical wire), but conducts up/down to other mese blocks.

**Comment by sofar** [comment]
Mesecons counts things as connected that are not visibly connected diagonally. The whole design for connected nodeboxes was to make coherent and complete visual connections, which is impossible to do diagonally without going outside of the nodebox.

I also think that mesecons' wire model is way too complex and a mistake. The amount of nodes needed for it surely is a design flaw, but even if you could do it in one node it is visually bizarre and inconsistent. For instance, there is no visual difference between these two situations:

Side view:

```
AW
WA
```
and
```
AW
WW
```

(A = air, W = mesecons wire)

**Comment by Jeija** [comment]
Please keep in mind that there are also several advantages of the current system over connected nodeboxes. Yes, startup takes longer, but mesecons circuits update a lot faster with the current nodebox system over connected nodes. That is because all those nodes have different connection rules, so that e.g. a straight wire only has 4 nodes that can be potentially connected to it, compared to 12 nodes that were previously necessary (when we had raillike nodes). Yes, we could technically store that information in the node's metadata; However, I generally prefer to have that information visible to make debugging easier and because the mesecons connnection rules system is generally based on nodenames.
Unless the engine has full support for meseconlike wires, I don't like to switch to any in-between solution. Developing an alternative wire (e.g. different color wires for insulated signals) without completely replacing the current wire system? I'd be all for that. We can then modularize the current wire so that server owners who don't want to have all those additional nodes can remove the default wire module.

**Comment by C1ffisme** [comment]
@Jeija Yeah, I knew  it would be a long process, which would break lots of things.

(Mumble mumble) Maybe if we ever get Meta setting Nodedef (Mumble mumble) :wink:

**Comment by sofar** [comment]
@Jeija indeed having e.g. a nice set of colored, insulated wires that connect to just the 6 faces would be trivial to add, and may be a huge improvement to gameplay - it will make circuits easily readable.

**Comment by C1ffisme** [comment]
@sofar Maybe a copper-based wire?

**Comment by sofar** [comment]
Given that old style wiring is nothing but copper wrapped in cotton, I'd totally support having copper insulated wire made out of either wool or cotton with copper in the middle.
